src/pages/add-edit/add-edit-page.jsx

```diff
@@ -67,239 +79,250 @@ export default class AddEditEventPage extends React.Component {
+            if (Array.isArray(eventData[key])) {
+              if (this.arraysEqual(eventData[key], this.state.seriesData[key])) {
+                delete eventData[key];
               }
-            });
-
-            url = `${window.abe_url}/events/${eventData.id || eventData.sid}/${eventData.recId}`;
-          } else {
-            url = `${window.abe_url}/events/${eventData.id || eventData.sid}`;
-          }
-          requestMethod = axios.put;
-        } else { // We're adding a new event
-          url = `${window.abe_url}/events/`;
-          requestMethod = axios.post;
+            } else if (moment.isMoment(eventData[key])) {
+              if (eventData[key].isSame(this.state.seriesData[key])) {
+                delete eventData[key];
+              }
+            }
+            if (eventData[key] === this.state.seriesData[key]) {
+              delete eventData[key]; // If so, don't override it for this event
+            }
+          });
+
+          url = `${window.abe_url}/events/${eventData.id || eventData.sid}/${this.props.match.params.recId}`;
+        } else {
+          url = `${window.abe_url}/events/${eventData.id || eventData.sid}`;
         }
-
-        // Make the request
-        requestMethod(url, eventData)
-          .then(this.props.eventSavedSuccessfully)
-          .catch(jqXHR => this.props.eventSaveFailed(eventData, jqXHR.message));
+        requestMethod = axios.put;
+      } else { // We're adding a new event
+        url = `${window.abe_url}/events/`;
+        requestMethod = axios.post;
       }
-    };
 
-    titleChanged = e => this.updateEventDatum({ title: e.currentTarget.value });
-
-    locationChanged = (loc) => {
-      // Save the processed/cleaned version to the eventData object
-      this.updateEventDatum({
-        location: loc.isOlin
-          ? [loc.building, loc.room, loc.suffix].join(' ').trim()
-          : loc.value.trim(),
-      });
-      // Save the dirty version to be passed on to the text field
-      this.setState({ locationRaw: loc.value });
-    };
-
-    allDayToggled = e => this.updateEventDatum({ allDay: e.currentTarget.checked });
-
-    setStart = (start) => {
-      const timeDelta = start.diff(this.state.eventData.start);
-      this.updateEventDatum({
-        start,
-        // Shift the end time to maintain the same duration
-        end: moment(this.state.eventData.end).add(timeDelta, 'ms'),
-      });
-    };
-
-    setEnd = end => this.updateEventDatum({ end });
-
-    doesRecurToggled = e => this.updateEventDatum({ doesRecur: e.currentTarget.checked });
-
-    recurrenceRuleChanged = recurrenceRule => this.setState({ recurrenceRule });
+      // Make the request
+      requestMethod(url, eventData)
+        .then(this.props.eventSavedSuccessfully)
+        .catch(jqXHR => this.props.eventSaveFailed(eventData, jqXHR.message));
+    }
+  };
 
-    descriptionChanged = description => this.updateEventDatum({ description });
+  arraysEqual = (arr1, arr2) => {
```

Comment:
The project already has a runtime dependency on lodash (because this is used by async, react-tag-input, input-moment, react-dnd, react-redux, redux, and react-tag-input), so you could also use [`_.isEqual`](https://lodash.com/docs/4.17.5#isEqual).

Reply:
Nifty...